x-pack/legacy/plugins/apm/server/lib/helpers/setup_request.ts

```diff
@@ -31,12 +32,17 @@ export async function setupRequest(req: Legacy.Request) {
   const query = (req.query as unknown) as APMRequestQuery;
   const { server } = req;
   const config = server.config();
+  const [uiFiltersES, indices] = await Promise.all([
+    decodeUiFilters(server, query.uiFilters),
+    getApmIndices(server)
```

Comment:
Currently `getApmIndices` is called twice for each request. First to get the list of APM indices, and afterwards to verify whether it is an APM index that was queries or not.

What do you think about caching `getApmIndices`  for a short amount of time, eg. 5 seconds?  That way if we make several concurrent requests hopefully they won't all have to call `getApmIndices`. At worst they'll only do it once per request.